include/unifex/transform_done.hpp

```diff
@@ -0,0 +1,364 @@
+  }
+
+private:
+  template<typename CPO, typename... Args>
+  friend auto tag_invoke(CPO cpo, const type& r, Args&&... args)
+      noexcept(std::is_nothrow_invocable_v<CPO, const Receiver&, Args...>)
+      -> std::invoke_result_t<CPO, const Receiver&, Args...> {
+    return std::move(cpo)(r.get_receiver(), (Args&&)args...);
+  }
+  
+  template <typename VisitFunc>
+  friend void tag_invoke(
+      tag_t<visit_continuations>,
+      const type& r,
+      VisitFunc&& func) noexcept(std::is_nothrow_invocable_v<
+                                VisitFunc&,
+                                const Receiver&>) {
+    std::invoke(func, r.get_receiver());
+  }
+
+  const Receiver& get_receiver() const noexcept {
+    assert(op_ != nullptr);   
+    return op_->receiver_;
+  }
+
+  operation* op_;
+};
+
+template<typename Source, typename Done, typename Receiver>
+class _op<Source, Done, Receiver>::type {
+  using source_receiver = receiver_type<Source, Done, Receiver>;
+  using final_receiver = final_receiver_type<Source, Done, Receiver>;
+
+public:
+  template<typename Source2, typename Done2, typename Receiver2>
+  explicit type(Source2&& source, Done2&& done, Receiver2&& dest)
+      noexcept(std::is_nothrow_move_constructible_v<Receiver> &&
+               std::is_nothrow_move_constructible_v<Done> &&
+               is_nothrow_connectable_v<Source2, source_receiver>)
+  : done_((Done2&&)done)
+  , receiver_((Receiver2&&)dest)
+  {
+    sourceOp_.construct_from([&] {
+        return unifex::connect((Source2&&)source, source_receiver{this});
+      });
+    startedOp_ = 0 + 1;
+  }
+
+  ~type() {
+    if (startedOp_ < 0) {
+      finalOp_.destruct();
+    } else if (startedOp_ > 0) {
+      sourceOp_.destruct();
+    }
+    startedOp_ = 0;
+  }
+
+  void start() & noexcept {
+    unifex::start(sourceOp_.get());
+  }
+
+private:
+  friend source_receiver;
+  friend final_receiver;
+
+  using source_op_t = operation_t<Source, source_receiver>;
+
+  using final_sender_t = std::invoke_result_t<Done&>;
+
+  using final_op_t = operation_t<final_sender_t, final_receiver>;
+
+  UNIFEX_NO_UNIQUE_ADDRESS Done done_;
+  UNIFEX_NO_UNIQUE_ADDRESS Receiver receiver_;
+  int startedOp_ = 0;
+  union {
+    manual_lifetime<source_op_t> sourceOp_;
+    manual_lifetime<final_op_t> finalOp_;
+  };
+};
+
+template<typename Source, typename Done>
+class _sndr<Source, Done>::type {
+  using final_sender_t = std::invoke_result_t<Done&>;
+
+public:
+  template<template<typename...> class Variant,
+           template<typename...> class Tuple>
+  using value_types = Variant<Tuple<>>;
+
+  template <template <typename...> class Variant>
+  using error_types = typename concat_type_lists_unique_t<
+      typename Source::template error_types<
+          decayed_tuple<type_list>::template apply>,
+      typename decltype(std::declval<Done>()())::template error_types<
+          decayed_tuple<type_list>::template apply>,
```

Comment:
Similarly here, don't need decayed_tuple here as the error values are just forwarded through.